new-chatbot-app/client/src/components/OfflineTicketWidget.jsx

```diff
@@ -0,0 +1,75 @@
+import { FormControl, FormLabel, Input, Button } from "@chakra-ui/react";
+import { useState, useContext } from "react";
+import { SocketContext } from "../context/SocketContextProvider";
+
+const OfflineTicketWidget = () => {
+
+  const [question, setQuestion] = useState('');
+  const [email, setEmail] = useState('');
+  const [name, setName] = useState('');
+  const [details, setDetails] = useState('');
+
+  const { socket } = useContext(SocketContext);
+
+  const handleTicketSubmit = (e) => {
+    e.preventDefault();
+    if (socket) {
+      socket.emit(
+        "createTicket",
+        {
+          question: question,
+          email: email,
+          name: name,
+          details: details,
+          ua: navigator.userAgent,
+        },
+        (responseMessage) => {
+          console.log(responseMessage);
+          setStep("initial");
+        }
+      );
+    }
+  };
```

Comment:
I think this should be in the SocketContextProvider, too. You can do something like:
function socketSendMessage(message) {
 socket.emit(
        "createTicket",
        {
          question: question,
          email: email,
          name: name,
          details: details,
          ua: navigator.userAgent,
        },
        (responseMessage) => {
          console.log(responseMessage);
          setStep("initial");
        }
      );
      }
      and then export this function for other Component to use easily